prompt/prompt-executor/prompt-executor-clients/prompt-executor-ollama-client/src/commonMain/kotlin/ai/koog/prompt/executor/ollama/client/dto/OllamaConverters.kt

```diff
@@ -11,12 +11,32 @@ import kotlinx.serialization.json.Json
 
 /**
  * Converts LLModel to Ollama model ID string.
+ * This function provides backward compatibility for hardcoded models.
+ * For dynamic model resolution, use OllamaModelResolver instead.
  */
 internal fun LLModel.toOllamaModelId(): String = when (this.id) {
     OllamaModels.Meta.LLAMA_3_2.id -> "llama3.2"
     OllamaModels.Alibaba.QWQ.id -> "qwq"
     OllamaModels.Alibaba.QWEN_CODER_2_5_32B.id -> "qwen2.5-coder:32b"
-    else -> error("Unknown profile ID ${this.id}")
+    else -> {
+        // Try to extract model name from dynamic model IDs
+        when {
+            this.id.startsWith("ollama-") -> {
+                val extractedName = this.id.removePrefix("ollama-")
+                if (extractedName.startsWith("dynamic-")) {
+                    extractedName.removePrefix("dynamic-")
+                } else {
+                    extractedName
+                }
+            }
+
+            else -> {
+                // Log warning and use ID directly as fallback
+                println("Warning: Unknown model ID ${this.id}, using ID directly. Consider using OllamaModelResolver for dynamic model support.")
+                this.id
+            }
+        }
+    }
```

Comment:
If we remove the `"ollama-"` and `"dynamic-"` prefixes and use the real Ollama model name as id, all this should be removed.